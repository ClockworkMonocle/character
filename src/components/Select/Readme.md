A simple wrapper around the HTML `<select></select>` element.

Basic usage:

```js
<Select>
  <option>Option 1</option>
  <option>Option 2</option>
  <option>Option 3</option>
  <option>Option 4</option>
  <option>Option 5</option>
</Select>
```

Usage with label:

```js
<Select label="Pick one">
  <option>Option 1</option>
  <option>Option 2</option>
  <option>Option 3</option>
  <option>Option 4</option>
  <option>Option 5</option>
</Select>
```
