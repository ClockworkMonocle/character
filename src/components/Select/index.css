
.select {
  display: block;
  position: relative;
}

.select_label {
  font-size: 0.70em;
  font-weight: bold;
  text-transform: uppercase;
  color: var(--attack-color);
}

.select select {
  display: block;
  width: 100%;
  font-size: 1em;
  padding: 8px;
  appearance: none;
  background-color: transparent;
  border: 0;
  border-bottom: 1px solid var(--black);
  border-radius: 0;
  outline: none;
  transition: border-color 300ms ease;
}

.select_indicator {
  display: block;
  position: absolute;
  right: 8px;
  top: 12px;
  height: 0;
  width: 0;
  content: ' ';
  border-left: 0.4em solid transparent;
  border-right: 0.4em solid transparent;
  border-top: 0.4em solid var(--black);
  transition: border-color 300ms ease;
}

.select_indicator--hasLabel {
  top: 30px;
}

.select select:active,
.select select:focus {
  border-color: var(--attack-color);
}

.select select:active ~ .select_indicator,
.select select:focus ~ .select_indicator {
  border-top-color: var(--attack-color);
}
