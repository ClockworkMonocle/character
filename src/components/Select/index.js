import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import './index.css';

export default function Select({ id, label, className, children, ...props }) {
  return (
    <label htmlFor={id} className={cn('select', className)}>
      {label && <span className="select_label">{label}</span>}
      <select id={id} {...props}>
        {children}
      </select>
      <span className={cn("select_indicator", label && "select_indicator--hasLabel")} />
    </label>
  );
}

Select.propTypes = {
  label: PropTypes.string,
};
