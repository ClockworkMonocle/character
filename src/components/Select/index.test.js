import React from 'react';
import { shallow } from 'enzyme';
import Select from './index';

test('<Select /> renders', () => {
  expect(shallow(<Select />)).toMatchSnapshot();
});

test('<Select /> renders with children', () => {
  const mount = shallow(<Select><option>1</option><option>2</option></Select>);
  expect(mount).toMatchSnapshot();
  expect(mount.find('option').length).toEqual(2);
});

test('<Select /> renders label', () => {
  expect(shallow(<Select label="my select box" />)).toMatchSnapshot();
});
