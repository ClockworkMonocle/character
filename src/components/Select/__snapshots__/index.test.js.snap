// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> renders 1`] = `
<label
  className="select"
>
  <select />
  <span
    className="select_indicator"
  />
</label>
`;

exports[`<Select /> renders label 1`] = `
<label
  className="select"
>
  <span
    className="select_label"
  >
    my select box
  </span>
  <select />
  <span
    className="select_indicator select_indicator--hasLabel"
  />
</label>
`;

exports[`<Select /> renders with children 1`] = `
<label
  className="select"
>
  <select>
    <option>
      1
    </option>
    <option>
      2
    </option>
  </select>
  <span
    className="select_indicator"
  />
</label>
`;
